include/boost/math/tools/differential_evolution.hpp

```diff
@@ -0,0 +1,261 @@
+      oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+      oss << ": F in (0, 1] is required, but got F=" << F << ".";
+      throw std::domain_error(oss.str());
+    }
+    if (max_generations_ < 1) {
+       oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+       oss << ": There must be at least one generation.";
+       throw std::invalid_argument(oss.str());
+    }
+  }
+
+  template <typename ArgumentContainer, class Func, class URBG>
+  ArgumentContainer
+  argmin(const Func cost_function, URBG &g,
+           ArgumentContainer* initial_guess= nullptr,
+           std::invoke_result_t<Func, ArgumentContainer> target_value = std::numeric_limits<std::invoke_result_t<Func, ArgumentContainer>>::quiet_NaN(),
+           std::atomic<bool> *cancellation = nullptr,
+           std::vector<std::pair<ArgumentContainer, std::invoke_result_t<Func, ArgumentContainer>>>
+               *queries = nullptr,
+          std::atomic<std::invoke_result_t<Func, ArgumentContainer>>* current_minimum_cost = nullptr
+         ) {
+    constexpr bool has_resize = requires(ArgumentContainer& t) {
+        t.resize(std::declval<typename ArgumentContainer::size_type>());
+    };
+    using ResultType = std::invoke_result_t<Func, ArgumentContainer>;
+    using std::clamp;
+    using std::round;
+    using std::isnan;
+    using Size_t = ArgumentContainer::size_type;
+    std::vector<ArgumentContainer> population(NP_);
+    for (size_t i = 0; i < population.size(); ++i) {
+      if constexpr (has_resize) {
+        // Resize it to same size as bounds_:
+        population[i].resize(bounds_.size());
+      } else {
+        // Argument type must be known at compile-time; like std::array:
+        if (population[i].size() != bounds_.size()) {
+          std::ostringstream oss;
+          oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+          oss << ": For containers which do not have resize, the default size must be the same as the dimension, ";
+          oss << "but the default container size is " << population[i].size() << " and the dimension of the problem is " << bounds_.size() << ".";
+          oss << " The function argument container type is " << typeid(ArgumentContainer).name() << ".\n";
+          throw std::runtime_error(oss.str());
+        }
+      }
+    }
+    // Why don't we provide an option to initialize with (say) a Gaussian distribution?
+    // > If the optimum's location is fairly well known,
+    // > a Gaussian distribution may prove somewhat faster, although it
+    // > may also increase the probability that the population will converge prematurely.
+    // > In general, uniform distributions are preferred, since they best reflect
+    // > the lack of knowledge about the optimum's location.
+    //  - Differential Evolution: A Practical Approach to Global Optimization
+    using std::uniform_real_distribution;
+    uniform_real_distribution<Real> dis(Real(0), Real(1));
```

Comment:
```suggestion
    std::uniform_real_distribution<Real> dis(Real(0), Real(1));
```

There should be no legal replacements for std::uniform_real_distribution